### PR TITLE
1st draft of typescript based backend

### PR DESCRIPTION
**Pull request by andreashille**

### CODE REVIEW

File: backend/sendmessage/src/login-user.ts [hunk 1]
```diff
@@ -18,7 +20,9 @@ const COHEN_SCALE = [
   '∞',
 ];
 
-async function loginUser(userId, groupId, config) {
+const EXPIRY_TIME_IN_HOUR = process.env.EXPIRY_TIME_IN_HOUR || '1';
```
Comment (siskavera) [inline backend/sendmessage/src/login-user.ts:23]: Here as well: when is this default needed?

File: backend/sendmessage/src/types.d.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import * as AWS from 'aws-sdk';
+
+export interface Config {
+  connectionId: string | undefined;
```
Comment (siskavera) [inline backend/sendmessage/src/types.d.ts:4]: Why not connectionId as optional parameter?

File: backend/onconnect/src/app.ts [hunk 3]
```diff
@@ -1,19 +1,23 @@
...
+import * as AWS from 'aws-sdk';
+import { APIGatewayProxyHandler } from 'aws-lambda/trigger/api-gateway-proxy';
 
 const ddb = new AWS.DynamoDB.DocumentClient({
   apiVersion: '2012-08-10',
   region: process.env.AWS_REGION,
 });
 
-exports.handler = async (event) => {
+const EXPIRY_TIME_IN_HOUR = process.env.EXPIRY_TIME_IN_HOUR || '1';
```
[the remark this answers is not in the document]
Comment (andreashille) [reply backend/onconnect/src/app.ts:9]: fallback if config is messed up

File: backend/sendmessage/src/login-user.ts (hunk 1 above)
Comment (andreashille) [reply backend/sendmessage/src/login-user.ts:23]: similar

**Review by siskavera — APPROVED** [review]
Sorry, this took a while but looks good to me :)

**Pull request merged**